src/librustc_mir/dataflow/move_paths/builder.rs

```diff
@@ -337,6 +369,7 @@ impl<'b, 'a, 'gcx, 'tcx> Gatherer<'b, 'a, 'gcx, 'tcx> {
                 }
                 if let Some((ref destination, _bb)) = *destination {
                     self.create_move_path(destination);
```

Comment:
This only initializes the destination on the non-panic-path, so you'll probably want some label on the `Init` you create to indicate that.

Reply:
fixed.